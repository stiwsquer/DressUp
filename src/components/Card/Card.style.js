import styled from "styled-components";
import { flex } from "../../GlobalStyles";
import Button from "../Button/Button";

export const StyledCard = styled.div`
  ${flex({ direction: "column" })}
  img {
    object-fit: cover;
    width: 100%;
    height: 100%;
  }
  padding: 1rem;
`;

export const FigCaption = styled.figcaption`
  position: absolute;
  bottom: 0;
  width: 100%;
  text-align: center;
  min-height: 10%;
  display: none;
  background: white;
`;

export const CardFigure = styled.figure`
  position: relative;
  .card-image-normal {
    display: block;
  }

  .card-image-alt {
    display: none;
  }
  &:hover {
    .card-image-alt {
      display: block;
    }

    .card-image-normal {
      display: none;
    }

    ${FigCaption} {
      display: block;
      button {
        &::after {
          content: "QUICK VIEW";
        }

        &:hover {
          opacity: 0.5;
        }
        display: block;
      }
    }
  }
`;

export const CardButton = styled(Button)`
  background: transparent;
  color: black;
  letter-spacing: 0.1rem;
  font-weight: normal;
  cursor: pointer;
  font-size: 1.5rem;
  position: absolute;
  top: 50%;
  right: 50%;
  transform: translateY(-50%) translateX(50%);
  padding: 0.5rem 1.5rem;
  border: 0.1rem solid black;
`;

export const CardTitle = styled.h4`
  font-size: 1.5rem;
  font-family: "Karla-200", sans-serif;
  margin: 2rem 0rem;
  &:hover {
    opacity: 0.5;
  }
`;

export const CardText = styled.p`
  font-size: 1.5rem;
  font-family: "Karla-800", sans-serif;
`;
