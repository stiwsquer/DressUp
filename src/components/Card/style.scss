@import "../../styles/global.scss";
.card {
  //   width: 25vw;
  display: flex;
  flex-direction: column;
  img {
    object-fit: cover;
    width: 100%;
    height: 100%;
  }
}

.fig-caption {
  position: absolute;
  bottom: 0;
  width: 100%;
  text-align: center;
  min-height: 10%;
  display: none;
  background: white;
  button {
    background: transparent;
    color: black;
    letter-spacing: 0.1rem;
    font-weight: normal;
    cursor: pointer;
    font-size: 1.5rem;
    position: absolute;
    top: 50%;
    right: 50%;
    transform: translateY(-50%) translateX(50%);
    padding: 0.5rem 1.5rem;
    border: 0.1rem solid black;
  }
}

.card-figure {
  position: relative;

  &:hover {
    .card-image-alt {
      display: block;
    }

    .card-image-normal {
      display: none;
    }

    .fig-caption {
      display: block;
      button {
        &::after {
          content: "QUICK VIEW";
        }

        &:hover {
          opacity: 0.5;
        }
        display: block;
      }
    }
  }
}

.card-image-normal {
  display: block;
}

.card-image-alt {
  display: none;
}

.card-body {
  h4 {
    font-size: $small-text;
    font-family: "Karla-200", sans-serif;
    margin: 2rem 0rem;
    &:hover {
      opacity: 0.5;
    }
  }
  p {
    font-size: $small-text;
    font-family: "Karla-800", sans-serif;
  }
}

.card-swatches {
  align-self: center;
  margin: 0.5rem;
  display: flex;
  .radio {
    input[type="radio"] {
      position: absolute;
      width: 2.5rem;
      height: 2.5rem;
      margin-left: 1rem;
      opacity: 0;
      cursor: pointer;
      + .radio-label {
        span {
          margin: 1rem;
          content: "";
          background: grey;
          border-radius: 100%;
          border: 0.1px solid rgba($color: #000000, $alpha: 0.3);
          display: inline-block;
          width: 2.5rem;
          height: 2.5rem;
          position: relative;
          top: -1rem;
          margin-right: 1rem;
          vertical-align: top;

          text-align: center;
          pointer-events: none;
          transition: all 0.25s ease;
        }
      }
      &:checked {
        + .radio-label {
          span {
            border: 0.3px solid rgba($color: black, $alpha: 0.5);
            box-shadow: 0 0 1rem rgba($color: red, $alpha: 0.6);
          }
        }
      }
      &:focus {
        + .radio-label {
          span {
            border: 0.3px solid rgba($color: black, $alpha: 0.5);
            box-shadow: 0 0 1rem rgba($color: red, $alpha: 0.6);
          }
        }
      }
      &:disabled {
        + .radio-label {
          span {
            box-shadow: inset 0 0 0 4px red;
            border-color: darken(red, 25%);
            background: darken(red, 25%);
          }
        }
      }
      + .radio-label {
        &:empty {
          span {
            margin-right: 0;
          }
        }
      }
    }
  }
}
