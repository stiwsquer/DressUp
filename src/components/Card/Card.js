import React from "react";
import "./style.scss";
import { Link } from "react-router-dom";
import { useState, useRef } from "react";
import Button from "../Button/Button";
import CardModal from "../CardModal/CardModal";

export default function Card({ linkTo, imagesSources, title, text }) {
  const [imgSrc, setImgSrc] = useState(imagesSources[0].src);
  const [imgSrcHover, setImgSrcHover] = useState(imagesSources[0].srcHover);
  const [imgAlt, setImgAlt] = useState(imagesSources[0].imgAlt);
  const [isModalOpen, setIsModalOpen] = useState(false);

  const handleInputRadioClick = (e) => {
    setImgSrc(imagesSources[e.target.value].src);
    setImgSrcHover(imagesSources[e.target.value].srcHover);
    setImgAlt(imagesSources[e.target.value].imgAlt);
  };

  return (
    <div className="card">
      <figure className="card-figure">
        <Link to={linkTo} className="card-image-link">
          <img
            className=" card-image card-image-normal"
            src={imgSrc}
            alt={imgAlt}
          />
          <img
            className="card-image  card-image-alt"
            src={imgSrcHover}
            alt={imgAlt}
          />
        </Link>
        <figcaption className="fig-caption">
          <Button
            onClick={() => {
              setIsModalOpen(true);
            }}
          ></Button>

          <CardModal
            open={isModalOpen}
            setIsModalOpen={setIsModalOpen}
            imagesSources={imagesSources}
            title={title}
            text={text}
          ></CardModal>
        </figcaption>
      </figure>
      <div className="card-body">
        <Link to={linkTo}>
          <h4 className="card-title">{title}</h4>
        </Link>
        <p className="card-text">{text}</p>
      </div>
      <div className="card-swatches">
        {imagesSources.map((source, index) => {
          return (
            <div key={index} className="radio">
              <input
                onClick={handleInputRadioClick}
                value={index}
                id="radio1"
                type="radio"
                name="swatch"
              ></input>
              <label htmlFor="" id="radio1" className="radio-label">
                <span
                  style={{
                    background: source.backgroundColor,
                  }}
                ></span>
              </label>
            </div>
          );
        })}
      </div>
    </div>
  );
}
