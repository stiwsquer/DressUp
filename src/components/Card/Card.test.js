import Card from "./Card";
import { Link } from "react-router-dom";

const imgSrc = [
  {
    src: "./img/products/01.0.jpg",
    srcHover: "./img/products/01.1.jpg",
    imgAlt: "Image of women in a dress",
    title: "Fiercely Fabulous Leopard Ruched Mini Dress",
    text: "$44.00",
    backgroundColor:
      'url("https://cdn11.bigcommerce.com/s-3y69bgeked/images/stencil/768w/attribute_value_images/73764.preview.jpg?t=1630492842)"',
  },
];

it("renders Card component without crashing", () => {
  shallow(<Card linkTo="#" imagesSources={imgSrc} />);
});

it("renders two <Link /> components", () => {
  const wrapper = shallow(<Card linkTo="#" imagesSources={imgSrc} />);
  expect(wrapper.find(Link)).toHaveLength(2);
});

it("contains h4 element", () => {
  const wrapper = shallow(<Card linkTo="#" imagesSources={imgSrc} />);
  expect(wrapper.find("h4")).toHaveLength(1);
});

it("Card snapshot", () => {
  const component = shallow(<Card linkTo="#" imagesSources={imgSrc} />);

  expect(component).toMatchSnapshot();

  component.find("figure").simulate("hover");

  expect(component).toMatchSnapshot();
});
