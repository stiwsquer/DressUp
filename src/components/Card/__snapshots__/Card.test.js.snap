// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card snapshot 1`] = `
<div
  className="card"
>
  <figure
    className="card-figure"
  >
    <Link
      className="card-image-link"
      to="#"
    >
      <img
        alt="Image of women in a dress"
        className=" card-image card-image-normal"
        src="./img/products/01.0.jpg"
      />
      <img
        alt="Image of women in a dress"
        className="card-image  card-image-alt"
        src="./img/products/01.1.jpg"
      />
    </Link>
    <figcaption
      className="fig-caption"
    >
      <Button
        onClick={[Function]}
      />
      <CardModal
        imagesSources={
          Array [
            Object {
              "backgroundColor": "url(\\"https://cdn11.bigcommerce.com/s-3y69bgeked/images/stencil/768w/attribute_value_images/73764.preview.jpg?t=1630492842)\\"",
              "imgAlt": "Image of women in a dress",
              "src": "./img/products/01.0.jpg",
              "srcHover": "./img/products/01.1.jpg",
              "text": "$44.00",
              "title": "Fiercely Fabulous Leopard Ruched Mini Dress",
            },
          ]
        }
        open={false}
        setIsModalOpen={[Function]}
      />
    </figcaption>
  </figure>
  <div
    className="card-body"
  >
    <Link
      to="#"
    >
      <h4
        className="card-title"
      />
    </Link>
    <p
      className="card-text"
    />
  </div>
  <div
    className="card-swatches"
  >
    <div
      className="radio"
      key="0"
    >
      <input
        id="radio1"
        name="swatch"
        onClick={[Function]}
        type="radio"
        value={0}
      />
      <label
        className="radio-label"
        htmlFor=""
        id="radio1"
      >
        <span
          style={
            Object {
              "background": "url(\\"https://cdn11.bigcommerce.com/s-3y69bgeked/images/stencil/768w/attribute_value_images/73764.preview.jpg?t=1630492842)\\"",
            }
          }
        />
      </label>
    </div>
  </div>
</div>
`;

exports[`Card snapshot 2`] = `
<div
  className="card"
>
  <figure
    className="card-figure"
  >
    <Link
      className="card-image-link"
      to="#"
    >
      <img
        alt="Image of women in a dress"
        className=" card-image card-image-normal"
        src="./img/products/01.0.jpg"
      />
      <img
        alt="Image of women in a dress"
        className="card-image  card-image-alt"
        src="./img/products/01.1.jpg"
      />
    </Link>
    <figcaption
      className="fig-caption"
    >
      <Button
        onClick={[Function]}
      />
      <CardModal
        imagesSources={
          Array [
            Object {
              "backgroundColor": "url(\\"https://cdn11.bigcommerce.com/s-3y69bgeked/images/stencil/768w/attribute_value_images/73764.preview.jpg?t=1630492842)\\"",
              "imgAlt": "Image of women in a dress",
              "src": "./img/products/01.0.jpg",
              "srcHover": "./img/products/01.1.jpg",
              "text": "$44.00",
              "title": "Fiercely Fabulous Leopard Ruched Mini Dress",
            },
          ]
        }
        open={false}
        setIsModalOpen={[Function]}
      />
    </figcaption>
  </figure>
  <div
    className="card-body"
  >
    <Link
      to="#"
    >
      <h4
        className="card-title"
      />
    </Link>
    <p
      className="card-text"
    />
  </div>
  <div
    className="card-swatches"
  >
    <div
      className="radio"
      key="0"
    >
      <input
        id="radio1"
        name="swatch"
        onClick={[Function]}
        type="radio"
        value={0}
      />
      <label
        className="radio-label"
        htmlFor=""
        id="radio1"
      >
        <span
          style={
            Object {
              "background": "url(\\"https://cdn11.bigcommerce.com/s-3y69bgeked/images/stencil/768w/attribute_value_images/73764.preview.jpg?t=1630492842)\\"",
            }
          }
        />
      </label>
    </div>
  </div>
</div>
`;
